### PR TITLE
Reduce boot time

### PR DESCRIPTION
**Pull request by mykaul**
1. Remove the grub boot menu in virt-sysprep phase
2. Remove the QEMU boot menu in domain template XML
3. Wait 5 seconds till the VM is potentially ready to
receive SSH connections

### CODE REVIEW

File: lago/prefix.py [hunk 1]
```diff
@@ -1225,6 +1226,7 @@ def _deploy_host(self, host):
                 return
 
             with LogTask('Wait for ssh connectivity'):
+                time.sleep(5)
```
Comment (david-caro) [inline lago/prefix.py]: This will make you wait 5 seconds every time you want to deploy the host no? Maybe we can do a first try before the sleep? It seems to me also that this belongs more into the `wait_for_ssh` function, or maybe to the start one if it's only needed when booting up the vm.

File: lago/sysprep.py [hunk 2]
```diff
@@ -117,6 +117,14 @@ def config_net_interface_dhcp(iface, hwaddr):
     )
 
 
+def edit(filename, expression):
```
Comment (david-caro) [inline lago/sysprep.py]: Nice

File: lago/sysprep.py [hunk 3]
```diff
@@ -117,6 +117,14 @@ def config_net_interface_dhcp(iface, hwaddr):
...
+    editstr = '%s:""%s""' % (filename, expression)
```
Comment (david-caro) [inline lago/sysprep.py]: Just to make sure, here there are two pairs of double quotes, is that what you intended?

File: lago/prefix.py (hunk 1 above)
Comment (mykaul) [inline lago/prefix.py]: In practice, no VM loads in less than 5 seconds, so why bother it with useless SSH requests?
You should not put it inside the wait_for_ssh(), as this wait can be used elsewhere. It's only on the very first call, when the VM loads.

File: lago/sysprep.py (hunk 3 above)
Comment (mykaul) [inline lago/sysprep.py]: Yes, it is. virt-edit specifically mentions that it should be used to ensure escaping.

File: lago/prefix.py (hunk 1 above)
Comment (david-caro) [inline lago/prefix.py]: Well, the issue is that the deploy command is not used only on boot, but also whenever you want no? It's a command of it's own.

Sorry I might not have explained well, let me rephrase, we should do one of:
   * Instead of doing the sleep first, do a first try, and then sleep (if keeping it in the deploy step).
   * Move the sleep to wherever we do the boot (probably the start step).

Comment (mykaul) [inline lago/prefix.py]: OK, so the caller of env_start essentially will do the sleep, before calling env_deploy?

Comment (mykaul) [inline lago/prefix.py]: @david-caro - see latest revision. I've removed the wait - moved it to ovirt-system-tests - after it starts the VM, ./run_suite.sh simply sleep for 5 seconds. Brings us the same effect, without touching the deploy part in Lago.

**Comment by mykaul** [comment]
ci test please

File: lago/vm.py [hunk 4]
```diff
@@ -94,7 +94,11 @@ def start(self):
         super(LocalLibvirtVMProvider, self).start()
         if not self.defined():
             with LogTask('Starting VM %s' % self.vm.name()):
-                self.libvirt_con.createXML(self._libvirt_xml())
+                dom = self.libvirt_con.createXML(self._libvirt_xml())
+                if not dom:
+                    raise RuntimeError(
+                        'Failed to create Domain: %s' % self._libvirt_xml()
+                    )
```
Comment (david-caro) [inline lago/vm.py]: This is how you ensure that its created! :) 👍

**Comment by david-caro** [comment]
Looks good to me!

File: lago/vm.py [hunk 5]
```diff
@@ -123,6 +127,10 @@ def bootstrap(self):
                         ),
                         sysprep.set_iscsi_initiator_name(self.vm.iscsi_name()),
                         sysprep.set_selinux_mode('enforcing'),
+                        sysprep.edit(
+                            "/boot/grub2/grub.cfg",
+                            "s/set timeout=5/set timeout=0/g"
+                        ) if self.vm.distro() == 'el7' else '',
```
Comment (nvgoldin) [inline lago/vm.py]: is there a reason to restrict it to el7 only? (at least on my laptop with fc24 it looks like the same substitution would work)

**Comment by mykaul** [comment]
ci test please

**Comment by mykaul** [comment]
ci merge please
